src/components/ConnectButton/ConnectButton.tsx

```diff
@@ -100,16 +115,29 @@ const ConnectButton: React.FC<ConnectButtonProps> = ({
         deactivateWeb3();
         if (isInitialized) logout();
     }
-
+    async function resolve() {
+        if (resolveAddress === true) {
+            let name = await web3?.lookupAddress(account!);
+            if (name !== null || undefined || '') {
+                setHasName(true);
+                setName(name!);
+            } else {
+                setHasName(false);
+            }
+        } else {
+            setHasName(false);
```

Comment:
instead of else just return after `setName();`

Reply:
```js
 if (name) {
                setName(name!);
                return setHasName(true);
            }
            return setHasName(false);
```
I did this, dont know if it is what you had in mind

i have to revert the state so it will render the address and not the name